rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -298,15 +297,26 @@ impl<TPayload: Clone + Send + Sync + 'static> StreamProcessor<TPayload> {
                 consumer_state.dlq_policy.flush(&request.positions);
                 self.consumer.commit_offsets(request.positions).unwrap();
             }
-            Err(e) => match self.buffered_messages.pop(&e.partition, e.offset) {
-                Some(msg) => {
-                    tracing::error!(?e, "Invalid message");
-                    consumer_state.dlq_policy.produce(msg);
-                }
-                None => {
-                    tracing::error!("Could not find invalid message in buffer");
+            Err(strategies::PollError::InvalidMessage(e)) => {
+                match self.buffered_messages.pop(&e.partition, e.offset) {
+                    Some(msg) => {
+                        tracing::error!(?e, "Invalid message");
+                        consumer_state.dlq_policy.produce(msg);
+                    }
+                    None => {
+                        tracing::error!("Could not find invalid message in buffer");
+                    }
                 }
-            },
+            }
+
+            Err(strategies::PollError::JoinError(error)) => {
+                let error: &dyn std::error::Error = &error;
+                tracing::error!(error, "the thread crashed");
```

Comment:
I think we still need to make sure that the consumer shuts down, so we still need to somehow return (or panic as a temporary measure)